README.md

```diff
@@ -1,39 +1,78 @@
 Instagram Challenge
 ===================
 
-## Instructions
+Build Instagram using Rails.
 
-* Feel free to use Google, your notes, books, etc., but work on your own
-* If you refer to the solution of another coach or student, please put a link to that in your README
-* If you have a partial solution, **still check in a partial solution**
-* You must submit a pull request to this repo with your code by 9am Monday morning
+[x] users can can post pictures
+[ ] users can write comments on pictures
+[ ] users can like a picture
+[ ] app is styled
 
-## Task
+```
+1. As a user  
+So that I can share my photos with other users  
+I want to to post a photo  
+
+2. As a user  
+So that I can say something about another users photo  
+I want to comment on a photo  
 
-Build Instagram: Simple huh!
+3. As a user  
+So that I can let another user know that I like their photo  
+I want to like a photo  
+```
+---
 
-Your challenge is to build Instagram using Rails. You'll need **users** who can post **pictures**, write **comments** on pictures and **like** a picture. Style it like Instagram's website (or more awesome).
 
-Bonus if you can add filters!
+#### set up tasks
+[x] initialize Rails  
+[x] add rubocop gems for code quality checks  
+[x] bundle  
 
-## How to start
+#### setting up a user account
+[x] Sign up functionality  
+  [x] bcrypt set up  
+  [x] Users model (database with username and password)  
+  [x] Form for signing up in new users view  
+  [x] User controller creates and saves user  
+[x] Log in functionality  
+  [x] Sessions set up:  
+    [x] Form for logging in in new sessions view  
+    [x] create sessions with user id  
+    [x] renders same page if no user found  
+    [x] current_user set by session user id, logged in username on root page  
+[x] Logout functionality  
+    [x] user clicks logout  
+    [x] destroy session  
 
-1. Produce some stories, break them down into tasks, and estimate
-2. Fork this repo, clone, etc
-3. Initialize a new rails project
+#### User story 1 tasks
+[x] share a photo button or link  
+  [x] form for uploading a photo  
+  [x] photo shows on the homepage posts feed    
 
-Remember to proceed in small steps! Getting confused? Make the steps even smaller.
 
-## Code Quality
+#### User story 2 tasks
+TBD  
 
-For linting, you can use the `.rubocop.yml` in this repository (or your own!).
-You'll need these gems:
+#### User story 3 tasks
+TBD  
+
+---
+
+### Current status and notes of project:
+
+![instagram](insta.png)  
+
+- run bin/rails s and visit http://localhost:3000/ for user testing  
```

Comment:
When I did this, got an error message from Rails when visiting the site:
```
Migrations are pending. To resolve this issue, run:
bin/rails db:migrate RAILS_ENV=development
```
After running that line in terminal, restarting the server and refreshing, unfortunately I got this error instead:
```
Webpacker::Manifest::MissingEntryError in Sessions#welcome
[...]/instagram-challenge/app/views/layouts/application.html.erb where line #10 raised:

Webpacker can't find application.js in /[...]/instagram-challenge/public/packs/manifest.json. Possible causes:
1. You want to set webpacker.yml value of compile to true for your environment
   unless you are using the `webpack -w` or the webpack-dev-server.
2. webpack has not yet re-run to reflect updates.
3. You have misconfigured Webpacker's config/webpacker.yml file.
4. Your webpack configuration is not creating a manifest.
Your manifest contains:
{
}
```

Reply:
I think you need to bundle and then bin/rails db:migrate? If that's all required i can add those commands to the readme too

Reply:
I ran `bundle install` initially, and then `bin/rails db:migrate RAILS_ENV=development` when I got that first error message; then also tried simply `bin/rails db:migrate` after getting the second error message, but still got the same error and wasn't sure what else to try - I'm a bit of a Rails n00b, but maybe one of the coaches would know what caused it.

Reply:
Interesting. I'll have to get my detective hat on soon. I don't fully understand webpacker, but i've noticed some people having trouble with that on their machines, but i also i see its mentioning the code itself in the error. Everything seems to run on my machine, so i wonder if it's something to do with webpacker itself, it's version? 

Also, maybe check that you're running it on the right version of ruby mentioned in the gemfile? and maybe there's a mismatch with rails version? I'm more drawn to it being a version thing on your side as it's working for me, but i'm also throwing guesses out there!

Reply:
I was running it on `ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [x86_64-darwin19]`, and `Rails 6.1.3.1`; and Webpacker is installed by default by Rails versions 6 & up - the version I have is `webpacker 5.2.1` (the Gemfile specifies ~>5.0).

Reply:
Yay i figured it out! Was able to replicate the error on a different machine. You will need to install yarn. I did `npm install --global yarn` then `bin/yarn`. It resolved the error and now the app works! (hope its the same resolve for you!)